frontend/src/pages/projects/screens/detail/storage/StorageTableRow.tsx

```diff
@@ -29,13 +42,28 @@ type StorageTableRowProps = {
 const StorageTableRow: React.FC<StorageTableRowProps> = ({
   obj,
   rowIndex,
+  storageClassConfigMap,
+  storageClassesLoaded,
+  setAlertStatuses,
   onDeletePVC,
   onEditPVC,
   onAddPVC,
 }) => {
   const [isExpanded, setExpanded] = React.useState(false);
   const isRootVolume = useIsRootVolume(obj);
 
+  const isStorageClassesAvailable = useIsAreaAvailable(SupportedArea.STORAGE_CLASSES).status;
+
+  const storageClass = storageClassConfigMap[obj.spec.storageClassName ?? ''];
+  const isStorageClassDisabled = storageClass && !storageClass.isEnabled;
+  const isStorageClassDeleted = storageClassesLoaded && !storageClass;
+  React.useEffect(() => {
+    setAlertStatuses((prevState) => ({
+      ...prevState,
+      [obj.metadata.name]: !!isStorageClassDisabled || isStorageClassDeleted,
+    }));
+  }, [isStorageClassDeleted, isStorageClassDisabled, obj.metadata.name, setAlertStatuses]);
+
```

Comment:
We should move all this alert logic back up a level because alert statuses is not even used in this row component.